movement/watch_faces/complication/endless_runner_face.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2024 <David Volovskiy>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef ENDLESS_RUNNER_FACE_H_
+#define ENDLESS_RUNNER_FACE_H_
+
+#include "movement.h"
+
+/*
+    ENDLESS_RUNNER face
+
+    This is a basic endless-runner, like the [Chrome Dino game](https://en.wikipedia.org/wiki/Dinosaur_Game).
+    On the title screen, you can select a difficulty by long-pressing LIGHT or toggle sound by long-pressing ALARM.
+    LED or ALARM are used to jump.
+    High-score is displayed on the top-right on the title screen. During a game, the current score is displayed.
+*/
+
+typedef struct {
+    // These are values that need saving between uses
+    uint32_t hi_score : 10;
+    uint32_t difficulty : 3;
+    bool soundOn;
+    uint32_t month_last_hi_score : 4;
+    uint32_t year_last_hi_score : 6;
+    uint32_t unused : 8;
+} endless_runner_state_t;
```

Comment:
The memory occupied by this structure can be safely reduced by 50% by reordering members and thereby getting rid of needless padding:

```C
typedef struct {
    uint32_t hi_score : 10;
    uint32_t difficulty : 3;
    uint32_t month_last_hi_score : 4;
    uint32_t year_last_hi_score : 6;
    bool soundOn;
} endless_runner_state_t;
```

Reply:
Even better:

```C
typedef struct {
    uint32_t hi_score : 10;
    uint32_t difficulty : 3;
    uint32_t month_last_hi_score : 4;
    uint32_t year_last_hi_score : 6;
    uint8_t soundOn : 1;
} endless_runner_state_t;
```

Reply:
Done